builtin/providers/aws/resource_aws_codecommit_trigger.go

```diff
@@ -0,0 +1,170 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codecommit"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCodeCommitTrigger() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCodeCommitTriggerCreate,
+		Read:   resourceAwsCodeCommitTriggerRead,
+		Delete: resourceAwsCodeCommitTriggerDelete,
+
+		Schema: map[string]*schema.Schema{
+			"repository_name": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) > 100 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 100 characters", k))
+					}
+					return
+				},
+			},
+
+			"trigger": &schema.Schema{
+				Type:     schema.TypeSet,
+				ForceNew: true,
+				Required: true,
+				MaxItems: 10,
```

Comment:
You can create up to 10 unique triggers in response to repository events, such as creating or deleting a branch.